### PR TITLE
Document chain joining

### LINKED ISSUE #74: Document chain joining in SDK
- [x] Mention it in key-concepts
- [x] API documentation for all 4 join actions

### PR DESCRIPTION
**Pull request by nicholeous**
- API docs for all 4 join actions
- Added joins to key concepts
- fixed typo in infrastructure docs

 closes #74

### CODE REVIEW

File: docs/gitbook/key-concepts/README.md [hunk 1]
```diff
@@ -195,3 +195,22 @@ var smartContract = {
 Because a smart contract must be deterministic a method of handling side effects on a blockchain must be used. Interbit uses the redux-saga API to manage side effects. If you require side effect management in your covenant simply export a `rootSaga` in your covenant package export in the same way you would export a redux saga.
 
 Refer to the [redux-saga](https://redux-saga.js.org/docs/introduction/BeginnerTutorial.html) documentation for implementation details.
+
+## Chain Joining
+
+Chains can join to eachother via a read or write join. These joins enable Interbit blockchains to share data and interact with eachother.
```
Comment (recurrence) [inline docs/gitbook/key-concepts/README.md]: eachother is two words

File: docs/gitbook/reference/interbit-covenant-utils/authorizeReceiveActions.md [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+When dispatched, this action allows a sender chain to send actions to it.
+
+This action should be dispatched to the receiver chain for a write join.
 
 #### Arguments
 
-1. `senderChainId` *(Object)*:
-2. `permittedActions` *(Object)*:
+1. `senderChainId` *(string)*: The chain ID of the blockchain that will send actions
+2. `permittedActions` *(array)*: The actions that the sender is permitted to send
```
Comment (recurrence) [inline docs/gitbook/reference/interbit-covenant-utils/authorizeReceiveActions.md]: array of what?

File: docs/gitbook/reference/interbit-covenant-utils/startConsumeState.md [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
+
+This action should be dispatched to the consumer chain for a read join.
 
 #### Arguments
 
-1. `provider` *(Object)*:
-2. `mount` *(Object)*:
-3. `joinName` *(Object)*:
+1. `provider` *(string)*: The chain ID of the blockchain that will be providing state
+2. `mount` *(array)*: A state path that the shared state will be mounted to
```
Comment (recurrence) [inline docs/gitbook/reference/interbit-covenant-utils/startConsumeState.md]: array of what?

File: docs/gitbook/reference/interbit-covenant-utils/startProvideState.md [hunk 4]
```diff
@@ -1,12 +1,16 @@
...
+
+This action should be dispatched to the provider chain for a read join.
 
 #### Arguments
 
-1. `consumer` *(Object)*:
-2. `statePath` *(Object)*:
-3. `joinName` *(Object)*:
+1. `consumer` *(string)*: The chain ID of the blockchain that will be consuming state
+2. `statePath` *(array)*: Defines the path to the state that will be shared with the consumer chain
```
Comment (recurrence) [inline docs/gitbook/reference/interbit-covenant-utils/startProvideState.md]: array of what?

File: docs/gitbook/key-concepts/README.md [hunk 5]
```diff
@@ -195,3 +195,22 @@ var smartContract = {
 Because a smart contract must be deterministic a method of handling side effects on a blockchain must be used. Interbit uses the redux-saga API to manage side effects. If you require side effect management in your covenant simply export a `rootSaga` in your covenant package export in the same way you would export a redux saga.
 
 Refer to the [redux-saga](https://redux-saga.js.org/docs/introduction/BeginnerTutorial.html) documentation for implementation details.
+
+## Chain Joining
+
+Chains can join to each other via a read or write join. These joins enable Interbit blockchains to share data and interact with eachother.
```
Comment (recurrence) [inline docs/gitbook/key-concepts/README.md]: missed an eachother :)

File: docs/gitbook/key-concepts/README.md [hunk 6]
```diff
@@ -195,3 +195,22 @@ var smartContract = {
...
+## Chain Joining
+
+Chains can join to each other via a read or write join. These joins enable Interbit blockchains to share data and interact with each other.
+
+## Read Joins
+
+A read join allows one blockchain to share application state with another. Both blockchains must agree to and authorize the join in order for any data to be shared.
+
+Once authorized on both sides, the providing blockchain will send verifyable blockheaders and data to the consuming chain which will receive a subset of the providing blockchain's state. A read join is authorized on the providing side with an [`'@@interbit/START_PROVIDE_STATE'`](../reference/interbit-covenant-utils/startProvideState.md) action and on the receiving side by an [`'@@interbit/START_CONSUME_STATE'`](../reference/interbit-covenant-utils/startConsumeState.md). Once both actions have been blocked the state will be shared.
+
```
Comment (djlmills) [inline docs/gitbook/key-concepts/README.md]: Terminology use `consuming side`, not `receiving side` to be consistent with the actions.

**Pull request merged**